usaspending_api/common/mixins.py

```diff
@@ -173,9 +173,11 @@ def filter_records(self, request, *args, **kwargs):
                 vector_sum += vector
             queryset = queryset.annotate(search=vector_sum)
 
+        subwhere = filters
         # Create structure the query so we don't need to use distinct
         # This happens by reforming the request as 'WHERE pk_id IN (SELECT pk_id FROM queryset WHERE filters)'
-        subwhere = Q(**{queryset.model._meta.pk.name + "__in": queryset.filter(filters).values_list(queryset.model._meta.pk.name, flat=True)})
+        if len(filters) > 0:
```

Comment:
Previously, we would construct this subquery even if there were no filters, which dramatically slowed down flat 'list' calls to an endpoint with no filters.